pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/FilterContext.java

```diff
@@ -25,12 +25,23 @@
 @Data
 public class FilterContext {
     private Subscription subscription;
-    private MessageMetadata msgMetadata;
+    private final MessageMetadata msgMetadata = new MessageMetadata();
+
+    public FilterContext() {
+    }
 
     public void reset() {
         subscription = null;
-        msgMetadata = null;
+        msgMetadata.clear();
     }
 
     public static final FilterContext FILTER_CONTEXT_DISABLED = new FilterContext();
+
+    public MessageMetadata getMsgMetadata() {
```

Comment:
This may be problematic because the filter may want to hold a reference to this object (I don't know why  but you know users sometimes do silly things)

Reply:
Please elaborate about the problem. What do you suggest as a solution?